### DISCUSSION TITLE
Building ZRX

### DISCUSSION THREAD

**Discussion by gswifort** [CEXT-Dan/PyRx#69] [Q&A]
Can I find a guide somewhere on how to build `PyRx.zrx` based on this repository?
Do I need Visual Studio for this? Is the community edition enough?
I don't want to fork yet, I'm looking forward to testing the changes you make.

**Comment by CEXT-Dan** [comment]
Have a look at BUILDENV.txt

-You will need the ZRX SDK
-You will need the wxWindows, it’s on my google drive so you won’t have to build it
-You will need BOOST 1_85_0, from the link in BUILDENV.txt, download boost_1_85_0-bin-msvc-all-32-64.7z
-You will need Python 

Once you have the requirements. You will need to add environment variables as described in BUILDENV.txt

![env](https://github.com/CEXT-Dan/PyRx/assets/26830697/c76c63d7-9154-407f-88a2-5a332132e923)

You will need visual studio 2022 (community is fine) with all the C++ packages installed. The ZRX SDK need 142 build tools. So you will need the to make sure the correct packages are installed.

![build env](https://github.com/CEXT-Dan/PyRx/assets/26830697/d6a1d69c-fe7c-4232-94d9-532cf0391a7a)

Important, the first time you open the PyRx.sln, it will ask you if you want to upgrade the projects, choose no.
Right click on PyRxZ25.0 and build it. You won’t be able to build the whole solution unless you have all the SDKs installed


![build](https://github.com/CEXT-Dan/PyRx/assets/26830697/3fe1d4b1-94e0-402d-9567-557a9ceec2ab)
